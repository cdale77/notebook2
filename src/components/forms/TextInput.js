import React from "react";
import PropTypes from "prop-types";

const TextInput = ({ title, name, fieldType, value, onChange }) => {
  return (
    <div className="form-group">
      <label htmlFor={name}>{title}</label>
      <input
        name={name}
        id={name}
        className={"input-" + name}
        type={fieldType || "text"}
        value={value}
        onChange={onChange}
      />
    </div>
  );
};

TextInput.propTypes = {
  onChange: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  fieldtype: PropTypes.string,
  value: PropTypes.string
};

export default TextInput;
