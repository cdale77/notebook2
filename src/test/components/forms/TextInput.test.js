import React from "react";
import Renderer from "react-test-renderer";
import TestUtils from "react-dom/test-utils";
import TestWrapper from "../../helpers/TestWrapper";
import TextInput from "../../../components/forms/TextInput";

describe("TextInput", () => {
  describe("rendering", () => {
    it("should render an input", () => {
      const component = Renderer.create(
        <TextInput title="title" name="name" value="foo" onChange={jest.fn()} />
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("should render an input with a default value", () => {
      const component = Renderer.create(
        <TextInput title="title" name="name" onChange={jest.fn()} />
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("should render an password input", () => {
      const component = Renderer.create(
        <TextInput
          title="title"
          name="name"
          fieldType="password"
          value="foo"
          onChange={jest.fn()}
        />
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("handling input", () => {
    let callBack, spy, component;

    beforeEach(() => {
      callBack = {
        onChange: () => {}
      };

      spy = jest.spyOn(callBack, "onChange");
      component = TestUtils.renderIntoDocument(
        <TestWrapper>
          <TextInput
            title="title"
            name="name"
            value="foo"
            onChange={callBack.onChange}
          />
        </TestWrapper>
      );
    });

    it("should fire the change function when the user types", () => {
      const field = TestUtils.findRenderedDOMComponentWithClass(
        component,
        "input-name"
      );

      field.value = "new value here";
      TestUtils.Simulate.change(field);

      expect(spy).toHaveBeenCalled();
      expect(callBack.onChange.mock.calls.length).toEqual(1);
    });
  });
});
