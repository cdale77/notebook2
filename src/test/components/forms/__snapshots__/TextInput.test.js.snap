// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput rendering should render an input 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="name"
  >
    title
  </label>
  <input
    className="input-name"
    id="name"
    name="name"
    onChange={[Function]}
    type="text"
    value="foo"
  />
</div>
`;

exports[`TextInput rendering should render an input with a default value 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="name"
  >
    title
  </label>
  <input
    className="input-name"
    id="name"
    name="name"
    onChange={[Function]}
    type="text"
    value={undefined}
  />
</div>
`;

exports[`TextInput rendering should render an password input 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="name"
  >
    title
  </label>
  <input
    className="input-name"
    id="name"
    name="name"
    onChange={[Function]}
    type="password"
    value="foo"
  />
</div>
`;
